discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -510,28 +537,10 @@ impl SearchScope for SqliteStorage {
 
     async fn get_document(&self, id: document::Id) -> Result<ApiDocumentView, Error> {
         let mut tx = self.pool.begin().await?;
-
-        let document = sqlx::query_as::<_, QueriedApiDocumentView>(
-            "SELECT
-                hd.documentId, nr.title, nr.snippet, nr.topic, nr.url, nr.image,
-                nr.datePublished, nr.source, nr.market, nc.domainRank, nc.score,
-                ur.userReaction, po.inBatchIndex, em.embedding, NULL AS stackId
-            FROM HistoricDocument       AS hd
-            JOIN NewsResource           AS nr   USING(documentId)
-            JOIN NewscatcherData        AS nc   USING(documentId)
-            JOIN PresentationOrdering   AS po   USING(documentId)
-            JOIN UserReaction           AS ur   USING(documentId)
-            JOIN Embedding              AS em   USING(documentId)
-            WHERE documentId = ?;",
-        )
-        .bind(id)
-        .fetch_one(&mut tx)
-        .await
-        .on_row_not_found(Error::NoDocument(id))?;
-
+        // It is correct that `HistoricDocument` instead of `SearchDocument` is used here.
```

Comment:
This function is in `storage.search().get_document()`.

So I initially expected it to return a document from the search, and in turn use the `SearchDocument` table as
base table (which is a subset of the `HistoricDocument` table, both tables can be seen as a set marking all documents which are in search/the history).

But this is used for getting documents for starting a deep search.

Which normally are search documents, but if it isn't it isn't a problem at all. So being a bit more flexible here is not a problem.

In the long term we probably could either change it to :
- use a generic get document from history function
-  return a smaller view  then the whole api document

But this can wait until we have done the migration, have remove the `storage` feature flag and and use the various db views in the engine instead of e.g. the current `document::Document` (as close to all code in the engine just needs a _very_ small subset of the fields).

I have clarified it :rocket: .